client/state/reader/follows/reducer.js

```diff
@@ -11,10 +11,83 @@ import {
 	READER_FOLLOW,
 	READER_UNFOLLOW,
 	READER_FOLLOWS_RECEIVE,
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_SUBSCRIBE_TO_NEW_COMMENT_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_COMMENT_EMAIL,
 } from 'state/action-types';
 import { prepareComparableUrl } from './utils';
 import { createReducer } from 'state/utils';
 
+function updatePostSubscription( state, { payload, type } ) {
+	const follow = find( state, { blog_ID: +payload.blogId } );
+	if ( ! follow ) {
+		return state;
+	}
+
+	const currentFollowState = get( follow, [ 'delivery_methods', 'email' ], {} );
+
+	const newProps = {
+		send_posts: ! ( type === READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL )
+	};
+
+	if ( payload.deliveryFrequency ) {
+		newProps.post_delivery_frequency = payload.deliveryFrequency;
+	}
+
+	const newFollowState = {
+		...currentFollowState,
+		...newProps,
+	};
+
+	if ( isEqual( currentFollowState, newFollowState ) ) {
```

Comment:
i'm not sure there is one given how the UI should work, at least not a common one. I had been thinking that the data-layer was going to kick off the same sub action twice, once by passing through the initial action and once when the request came back, but we ditched that in favor of using update instead. 

So yeah, I guess this won't really happen, unless you dispatch update with the same params by hand?